src/citrine/resources/design_workflow.py

```diff
@@ -17,15 +17,23 @@ class DesignWorkflowCollection(Collection[DesignWorkflow]):
     _collection_key = 'response'
     _resource = DesignWorkflow
 
-    def __init__(self, project_id: UUID, session: Session):
+    def __init__(self, project_id: UUID, branch_id: UUID, session: Session):
         self.project_id: UUID = project_id
+        self.branch_id: UUID = branch_id
         self.session: Session = session
 
+        self._api_version = "v1" if self.branch_id is None else "v2"
+
+    def register(self, model: DesignWorkflow) -> DesignWorkflow:
+        model.branch_id = self.branch_id
+        return super().register(model)
+
     def build(self, data: dict) -> DesignWorkflow:
         """Build an individual DesignExecution."""
         workflow = DesignWorkflow.build(data)
         workflow._session = self.session
         workflow.project_id = self.project_id
+        workflow.branch_id = self.branch_id
```

Comment:
I think we should be able to get this for free from the response body since you've added it here: https://github.com/CitrineInformatics/citrine-python/pull/702/files#diff-85cbb32077676199db6ca0d43596047d4b3956d22f1d1a055fd28ea3872d1eb5R35

Reply:
Ah, thank you! I'd meant to try this out, and clearly forgot. Just tried now (register, get, and list), and all populated it properly.